src/main/java/com/fasterxml/jackson/databind/deser/std/UntypedObjectDeserializer.java

```diff
@@ -508,6 +548,8 @@ protected Object mapObject(JsonParser p, DeserializationContext ctxt) throws IOE
             LinkedHashMap<String, Object> result = new LinkedHashMap<>(2);
             result.put(key1, value1);
             return result;
+        } else if (_customKeyDeserializer != null) {
+            key2 = (String) _customKeyDeserializer.deserializeKey(key2, ctxt);
```

Comment:
@JooHyukKim  I think same check needed for 2 cases further down too?
(and test should check that first 5 keys per level get custom deserialized as well)

Reply:
Yes, seems like it. Will do

Reply:
Applied ur review and did minor plummeting (with a new helper function) via https://github.com/FasterXML/jackson-databind/pull/4684/commits/1f7dfbc4dfa08a6b936e03e866f3ecfe6fc8c58c. Thanks!